### ISSUE TITLE
rbenv's homebrew formula is broken for --HEAD installs

### ISSUE THREAD

**Issue by jasonkarns** [rbenv/rbenv#1474]
Opening this because I don't have the time yet to dig into it...

Installation of the homebrew formula using --HEAD currently fails:

```
$ HOMEBREW_NO_INSTALL_FROM_API=1 brew install rbenv --HEAD
==> Fetching dependencies for rbenv: ruby-build
==> Fetching ruby-build
==> Downloading https://ghcr.io/v2/homebrew/core/ruby-build/manifests/20230202
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/ruby-build/blobs/sha256:3dbf454935a5bd36e3f9cc64e1b8d1f3404d289123eda4877a9179474d2abeae
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:3dbf454935a5bd36e3f9cc64e1b8d1f3404d289123eda4877a9179474d2abeae?se=2023-02-03T19%3A0
######################################################################## 100.0%
==> Fetching rbenv
==> Cloning https://github.com/rbenv/rbenv.git
Cloning into '/Users/jasonkarns/.cache/Homebrew/rbenv--git'...
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
==> Installing dependencies for rbenv: ruby-build
==> Installing rbenv dependency: ruby-build
==> Pouring ruby-build--20230202.all.bottle.tar.gz
🍺  /opt/homebrew/Cellar/ruby-build/20230202: 588 files, 300.5KB
==> Installing rbenv --HEAD
Error: inreplace failed
libexec/rbenv:
  expected replacement of "\"${BASH_SOURCE%/*}\"/../libexec" with #<Pathname:/opt/homebrew/Cellar/rbenv/HEAD-61747c0/libexec>

READ THIS: https://docs.brew.sh/Troubleshooting

rbenv's formula was built from an unstable upstream --HEAD.
This build failure is expected behaviour.
Do not create issues about this on Homebrew's GitHub repositories.
Any opened issues will be immediately closed without response.
Do not ask for help from MacHomebrew on Twitter.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

I assume this project owns the changes to rbenv's formula.

**Comment by chenrui333** [comment]
- 👋 I have just filed a PR on this issue, https://github.com/Homebrew/homebrew-core/pull/122918, but I am still seeing the rbenv test failure as I have seen in https://github.com/Homebrew/homebrew-core/pull/122671. 

@mislav any idea why the `shell integration` is broken?

**Comment by carlocab** [comment]
Fixed in Homebrew/homebrew-core#122918.

**Comment by jasonkarns** [comment]
thank you @carlocab and @chenrui333 !!
